pennylane/measurements/classical_shadow.py

```diff
@@ -314,6 +315,115 @@ def process(self, tape, device):
 
         return qml.math.cast(qml.math.stack([outcomes, recipes]), dtype=np.int8)
 
+    def process_state_with_shots(
```

Comment:
So external devices (like the AWS on-demand simulators, or QPUs) would to have to implement their own versions of this method? Is there any way it can be broken down into smaller components to make that easier?